source/common/config/ttl.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <chrono>
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+
+#include "absl/container/btree_set.h"
+
+namespace Envoy {
+namespace Config {
+
+/**
+ * Class for managing TTL expiration of xDS resources. TTLs are managed with a single timer that
+ * will always be set to the time until the next TTL, ensuring that we have a constant amount of
+ * timers per xDS resource type.
```

Comment:
Can you add some comments somewhere on the tradeoffs made in data structures here? I'm guessing a B-tree ordered set is used for efficient insertion/removal and reasonable prefix traversal, but could be good to discuss.

Reply:
We actually use std::set in order to get iterator stability. I've added a comment with some justifications, lmk what you think